api/v1beta1/topicpermission_webhook_test.go

```diff
@@ -0,0 +1,144 @@
+package v1beta1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("topic permission webhook", func() {
+	var permission = TopicPermission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: TopicPermissionSpec{
+			User:  "test-user",
+			Vhost: "/a-vhost",
+			Permissions: TopicPermissionConfig{
+				Exchange: "a-exchange",
+				Read:     ".*",
+				Write:    ".*",
+			},
+			RabbitmqClusterReference: RabbitmqClusterReference{
+				Name: "a-cluster",
+			},
+		},
+	}
+
+	Context("ValidateCreate", func() {
+		It("does not allow user and userReference to be specified at the same time", func() {
+			invalidPermission := permission.DeepCopy()
+			invalidPermission.Spec.UserReference = &corev1.LocalObjectReference{Name: "invalid"}
+			invalidPermission.Spec.User = "test-user"
+			Expect(apierrors.IsInvalid(invalidPermission.ValidateCreate())).To(BeTrue())
+		})
+		It("does not allow both user and userReference to be unset", func() {
+			invalidPermission := permission.DeepCopy()
+			invalidPermission.Spec.UserReference = nil
+			invalidPermission.Spec.User = ""
+			Expect(apierrors.IsInvalid(invalidPermission.ValidateCreate())).To(BeTrue())
+		})
+
+		It("does not allow both spec.rabbitmqClusterReference.name and spec.rabbitmqClusterReference.connectionSecret be configured", func() {
```

Comment:
We were noticing that there are not reference of "connectionSecret" in topic_permission_webhook.go, where is this situation managed?

Reply:
It's validated as a method on `RabbitmqClusterReference` so it's shared across all webhooks: https://github.com/rabbitmq/messaging-topology-operator/blob/topic-permission-191/api/v1beta1/topicpermission_webhook.go#L41